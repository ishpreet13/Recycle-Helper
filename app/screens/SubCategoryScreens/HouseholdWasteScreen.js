import React, { useEffect } from "react";
import { View, StyleSheet, Text } from "react-native";
import { BackHandler } from "react-native";

import AppButton from "../../components/AppButton";
import Screen from "../../components/Screen";
import Styles from "../../config/Styles";
import colors from "../../config/colors";

function HouseholdWasteScreen({ navigation }) {
  const handleBackButtonClick = () => {
    navigation.navigate("MainCategoryList");
    return true;
  };
  useEffect(() => {
    BackHandler.addEventListener("hardwareBackPress", handleBackButtonClick);
    return () => {
      BackHandler.removeEventListener(
        "hardwareBackPress",
        handleBackButtonClick
      );
    };
  }, []);
  return (
    <Screen>
      <View style={styles.container}>
        <View style={styles.heading}>
          <Text style={Styles.heading}>
            Select Household Waste Category for Recycle Instructions
          </Text>
        </View>
        <View style={styles.buttonContainer}>
          <AppButton
            title="CFLs"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "CFLs",
              })
            }
          ></AppButton>
          <AppButton
            title="Fluorescent Tubes"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Fluorescent Tube",
              })
            }
          ></AppButton>
          <AppButton
            title="Mercury Containing Items"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Mercury Containing Items",
              })
            }
          ></AppButton>
          <AppButton
            title="Medical Sharps"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Medical Sharps",
              })
            }
          ></AppButton>
          <AppButton
            title="Medications"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Medications",
              })
            }
          ></AppButton>
          <AppButton
            title="Paint"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Paint",
              })
            }
          ></AppButton>
          <AppButton
            title="Pesticides and Containers"
            onPress={() =>
              navigation.navigate("InstructionScreen", {
                documentName: "Pesticides and Containers",
              })
            }
          ></AppButton>
        </View>
      </View>
    </Screen>
  );
}

const styles = StyleSheet.create({
  buttonContainer: {
    width: "90%",
  },
  container: {
    flex: 1,
    alignItems: "center",
    backgroundColor: colors.white,
  },
  heading: {
    marginTop: 20,
    marginBottom: 20,
    paddingLeft: 2,
    paddingRight: 2,
  },
});
export default HouseholdWasteScreen;
