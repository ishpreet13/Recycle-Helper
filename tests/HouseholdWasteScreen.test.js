import "react-native";
import React from "react";
import renderer, { act } from "react-test-renderer";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

import HouseholdWasteScreen from "../app/screens/SubCategoryScreens/HouseholdWasteScreen";

act(() => {
  describe("Testing HouseholdWasteScreen ", () => {
    let wrapper;
    it("should render", async () => {
      act(() => {
        wrapper = renderer.create(<HouseholdWasteScreen />);
      });
      expect(wrapper.toJSON()).toBeTruthy();
    });
  });
});

act(() => {
  it("HouseholdWasteScreen should render correctly", async () => {
    let tree;
    act(() => {
      tree = renderer.create(<HouseholdWasteScreen />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("HouseholdWasteScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <HouseholdWasteScreen navigation={{ navigate }} />
    );
    fireEvent.press(getByText("CFLs"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "CFLs",
    });
  });
});

describe("HouseholdWasteScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <HouseholdWasteScreen navigation={{ navigate }} />
    );
    fireEvent.press(getByText("Fluorescent Tubes"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Fluorescent Tube",
    });
  });
});

describe("HouseholdWasteScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <HouseholdWasteScreen navigation={{ navigate }} />
    );
    fireEvent.press(getByText("Mercury Containing Items"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Mercury Containing Items",
    });
  });
});

describe("HouseholdWasteScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <HouseholdWasteScreen navigation={{ navigate }} />
    );
    fireEvent.press(getByText("Medical Sharps"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Medical Sharps",
    });
  });
});

describe("HouseholdWasteScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <HouseholdWasteScreen navigation={{ navigate }} />
    );
    fireEvent.press(getByText("Medications"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Medications",
    });
  });
});

describe("HouseholdWasteScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <HouseholdWasteScreen navigation={{ navigate }} />
    );
    fireEvent.press(getByText("Paint"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Paint",
    });
  });
});

describe("HouseholdWasteScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <HouseholdWasteScreen navigation={{ navigate }} />
    );
    fireEvent.press(getByText("Pesticides and Containers"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Pesticides and Containers",
    });
  });
});
