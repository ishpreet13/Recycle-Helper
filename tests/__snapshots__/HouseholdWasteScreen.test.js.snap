// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseholdWasteScreen should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
          "marginTop": 20,
          "paddingLeft": 2,
          "paddingRight": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "yellowgreen",
            "fontFamily": "Avenir",
            "fontSize": 22,
            "fontWeight": "bold",
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Select Household Waste Category for Recycle Instructions
      </Text>
    </View>
    <View
      style={
        Object {
          "width": "90%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          CFLs
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Fluorescent Tubes
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Mercury Containing Items
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Medical Sharps
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Medications
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Paint
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "yellowgreen",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Pesticides and Containers
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
